src/main/java/com/sap/sgs/phosphor/fosstars/model/score/WeightedCompositeScore.java

```diff
@@ -124,14 +124,14 @@ public final ScoreValue calculate(Value... values) {
     ScoreValue scoreValue = new ScoreValue(this);
     boolean allNotApplicable = true;
     for (Score subScore : subScores) {
-      Weight weight = weights.of(subScore).orElseThrow(IllegalStateException::new);
       ScoreValue subScoreValue = calculateIfNecessary(subScore, valueSet);
+      scoreValue.usedValues(subScoreValue);
```

Comment:
Here also the same concern. If the subscore value is not applicable. Should it be used?

Reply:
Please see above.